cmd/thor/p2p/p2p.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package p2p
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/nat"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/pkg/errors"
+	"github.com/vechain/thor/v2/comm"
+	"github.com/vechain/thor/v2/p2psrv"
+)
+
+type P2P struct {
+	comm           *comm.Communicator
+	p2pSrv         *p2psrv.Server
+	peersCachePath string
+	enode          string
+}
+
+func New(
+	communicator *comm.Communicator,
+	privateKey *ecdsa.PrivateKey,
+	instanceDir string,
+	nat nat.Interface,
+	version string,
+	maxPeers int,
+	listenPort int,
+	listenAddr string,
+	allowedPeers []*discover.Node,
+	cachedPeers []*discover.Node,
+	bootstrapNodes []*discover.Node,
+) *P2P {
+	// known peers will be loaded/stored from/in this file
+	peersCachePath := filepath.Join(instanceDir, "peers.cache")
+
+	// default option setting
+	// no known nodes for p2p connection
+	// use the hardcoded fallbackDiscoveryNodes for discovery only
+	opts := &p2psrv.Options{
+		Name:                common.MakeName("thor", version),
+		PrivateKey:          privateKey,
+		MaxPeers:            maxPeers,
+		ListenAddr:          listenAddr,
+		DiscoveryNodes:      fallbackDiscoveryNodes,
+		RemoteDiscoveryList: remoteDiscoveryNodesList,
+		NAT:                 nat,
+	}
+
+	// allowed peers flag will only allow p2psrv to connect to the designated peers
+	if len(allowedPeers) > 0 {
+		opts.NoDiscovery = true // disable discovery
+		opts.DiscoveryNodes = nil
+		opts.KnownNodes = allowedPeers
+	} else {
+		// boot nodes flag will overwrite the default bootstrap nodes and also disable remote bootstrap
+		if len(bootstrapNodes) > 0 {
+			opts.RemoteDiscoveryList = "" // disable remote bootstrap
+			opts.DiscoveryNodes = nil
```

Comment:
```suggestion
			opts.DiscoveryNodes = bootstrapNodes
```
It's good to have this line indication bootstrap nodes will also be used as discovery nodes